File: app/src/main/java/fr/erictruong/material/training/ui/layout/LayoutActivity.java
package fr.erictruong.material.training.ui.layout;

import android.support.v4.app.Fragment;
import android.content.Context;
import android.content.Intent;

import fr.erictruong.material.training.R;
import fr.erictruong.material.training.ui.core.BaseDrawerActivity;
import fr.erictruong.material.training.ui.core.DummyFragment;

public class LayoutActivity extends BaseDrawerActivity {

    /**
     * Start an {@link LayoutActivity} with extras.
     *
     * @param context The Android context.
     * @param id      (Extra) The navigation menu item ID of the desired content.
     */
    public static void start(Context context, int id) {
        Intent starter = new Intent(context, LayoutActivity.class);
        starter.putExtra(EXTRA_NAV_ID, id);
        context.startActivity(starter);
    }

    @Override
    protected Fragment getSelectedFragment(int id) {
        switch (id) {
            case R.id.nav_principles:
                return DummyFragment.newInstance();
            case R.id.nav_units_and_measurements:
                return DummyFragment.newInstance();
            case R.id.nav_metrics_and_keylines:
                return DummyFragment.newInstance();
            case R.id.nav_structure:
                return DummyFragment.newInstance();
            case R.id.nav_adaptive_ui:
                return DummyFragment.newInstance();
            default:
                return null;
        }
    }
}

File: app/src/main/java/fr/erictruong/material/training/ui/layout/LayoutActivity.kt
package fr.erictruong.material.training.ui.layout

import android.support.v4.app.Fragment
import fr.erictruong.material.training.R
import fr.erictruong.material.training.ui.core.BaseDrawerActivity
import fr.erictruong.material.training.ui.layout.metrics_and_keylines.MetricsAndKeylinesFragment
import fr.erictruong.material.training.ui.layout.principles.PrinciplesFragment
import fr.erictruong.material.training.ui.layout.responsive_ui.ResponsiveUiFragment
import fr.erictruong.material.training.ui.layout.split_screen.SplitScreenFragment
import fr.erictruong.material.training.ui.layout.structure.StructureFragment
import fr.erictruong.material.training.ui.layout.units_and_measurements.UnitsAndMeasurementsFragment
import org.jetbrains.anko.AnkoLogger
import org.jetbrains.anko.debug

class LayoutActivity : BaseDrawerActivity(), AnkoLogger {

    override fun getSelectedFragment(id: Int): Fragment? {
        when (id) {
            R.id.nav_principles -> return PrinciplesFragment()
            R.id.nav_units_and_measurements -> return UnitsAndMeasurementsFragment()
            R.id.nav_metrics_and_keylines -> return MetricsAndKeylinesFragment()
            R.id.nav_structure -> return StructureFragment()
            R.id.nav_responsive_ui -> return ResponsiveUiFragment()
            R.id.nav_split_screen -> return SplitScreenFragment()
            else -> {
                debug("Unknown ID: $id")
                return null
            }
        }
    }
}
